File: containerwithmostwater.js
var maxArea = function(height) {
    let left = 0, right = height.length - 1;
    let maxArea = 0;

    while (left < right) {
        let area = (right - left) * Math.min(height[left], height[right]);
        maxArea = Math.max(maxArea, area);

        // Move the pointer with the shorter height
        if (height[left] < height[right]) {
            left++;
        } else {
            right--;
        }
    }

    return maxArea;
};

// Example usage
console.log(maxArea([1,8,6,2,5,4,8,3,7]));  // Output: 49

File: containerwithmostwater.py
from typing import List

class Solution:
    def maxArea(self, height: List[int]) -> int:
        left, right = 0, len(height) - 1
        max_area = 0

        while left < right:
            # Calculate the current area
            area = (right - left) * min(height[left], height[right])
            max_area = max(max_area, area)
            
            # Move the pointer at the shorter height
            if height[left] < height[right]:
                left += 1
            else:
                right -= 1

        return max_area

# Example usage
sol = Solution()
print(sol.maxArea([1,8,6,2,5,4,8,3,7]))  # Output: 49
